test/test_utils/__init__.py

```diff
@@ -650,11 +650,30 @@ def get_framework_and_version_from_tag(image_uri):
             f"Cannot find framework in image uri {image_uri} " f"from allowed frameworks {allowed_frameworks}"
         )
 
-    tag_framework_version = image_uri.split(":")[-1].split("-")[0]
+    tag_framework_version = re.search(":(.*?)-", image_uri).groups()[0]
 
     return tested_framework, tag_framework_version
 
 
+def get_cuda_version_from_tag(image_uri):
+    """
+    Return the cuda version from the image tag.
+    :param image_uri: ECR image URI
+    :return: cuda version
+    """
+    cuda_framework_version = None
+
+    cuda_str = ["cu", "gpu"]
+    if all(keyword in image_uri for keyword in cuda_str):
+        cuda_framework_version = re.search("(cu.*?)-", image_uri).groups()[0]
```

Comment:
nit: Could you change the expression from `"cu.*?"` to `"cu\d{3}"?

Also, please use raw strings for regular expressions, by placing an `r` before the string literal. For example, `r"(cu\d{3})-"`

Reply:
Addressed. Function should also support other cuda versions (say cu92), change includes support for this kind of extensibility